code/game/machinery/doors/airlock.dm

```diff
@@ -117,6 +118,28 @@
 		max_integrity = normal_integrity
 	if(damage_deflection == AIRLOCK_DAMAGE_DEFLECTION_N && security_level > AIRLOCK_SECURITY_METAL)
 		damage_deflection = AIRLOCK_DAMAGE_DEFLECTION_R
+	if(abandoned)
+		var/outcome = rand(0,100)
+		switch(outcome)
```

Comment:
Why not `switch(rand(1, 100)`? You might want to replace the `0, 100` with `1, 100` either way, so it can't roll a 0 by accident.

Reply:
I always forget whether this shit is inclusive.